CineCompass/Model/MovieManager.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  MovieManager.swift
+//  CineCompass
+//
+//  Created by Muhammet BOZKURT on 20.04.2023.
+//
+
+import Foundation
+
+final class MovieManager {
+    let searchURL = "https://api.themoviedb.org/3/search/movie"
+    let apiKey = "replace this text with your api key"
+    
+    func searchMovies(movieName: String) {
+        let urlString = "\(searchURL)?api_key=\(apiKey)&query=\(movieName)"
+        // create url
+        guard let url = URL(string: urlString) else {
+            print("Invalid URL")
+            return
+        }
+        // create urlsession
+        let session = URLSession(configuration: .default)
+        // create url session a task
+        let task = session.dataTask(with: url) { data, response, error in
```

Comment:
sana tavsiyem bu [article](https://www.appypie.com/how-to-fix-strong-reference-cycle-swift) da ne varsa oku, ama çok konu var. O yüzden şimdilik bu basliktaki kısma bak "A Closure With “self” Reference Cycle"

ve su satira bu [weak self] ekleyerek düzelt. Bu strong reference cycle i kiracak ve memory leak olmasının engelleyecektir.

`let task = session.dataTask(with: url) { data, response, error in`
to
`let task = session.dataTask(with: url) { [weak self] data, response, error  in`

bunu yapinca compliler self. kisminda bir erroe verecek onu da duzelt

Reply:
bu degisiklikten sonra output alamamaya başladım